process-cpp: Refactor & update

OPNA2608 opened the pull request:
## Description of changes

- Add me to maintainers
- Move to by-name
- Switch to `finalAttrs` pattern
- Switch to `src.hash`
- Add `pkgConfigModules` & test
- Add `updateScript`
- unstable-2021-05-11 -> 3.0.1-unstable-2024-03-14
- Enable tests
- Enable `strictDeps`
- Add outputs

There's no changelog, but the [latest changes](https://gitlab.com/ubports/development/core/lib-cpp/process-cpp/-/commits/main/?ref_type=HEADS) are more or less just updating the C++ standard & fixing the tests on newer Linux. I think backporting this to 24.05 should be fine.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

OPNA2608 committed:
process-cpp: Add OPNA2608 to maintainers

OPNA2608 committed:
process-cpp: Move to by-name

OPNA2608 commented:
Lomiri is unbroken on master now, so:

---

Result of `nixpkgs-review pr 313759` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>dbus-cpp</li>
    <li>dbus-cpp.dev</li>
    <li>dbus-cpp.doc</li>
    <li>dbus-cpp.examples</li>
    <li>lomiri.biometryd</li>
    <li>lomiri.biometryd.dev</li>
    <li>lomiri.lomiri</li>
    <li>lomiri.lomiri-session</li>
    <li>lomiri.lomiri-system-settings</li>
    <li>lomiri.lomiri-system-settings-security-privacy</li>
    <li>lomiri.mediascanner2</li>
    <li>lomiri.mediascanner2.dev</li>
    <li>lomiri.qtmir</li>
    <li>lomiri.qtmir.dev</li>
    <li>lomiri.trust-store</li>
    <li>lomiri.trust-store.bin</li>
    <li>lomiri.trust-store.dev</li>
    <li>lomiri.trust-store.doc</li>
    <li>net-cpp</li>
    <li>net-cpp.dev</li>
    <li>net-cpp.doc</li>
    <li>process-cpp</li>
    <li>process-cpp.dev</li>
  </ul>
</details>

---

@onny Any objections?

@ofborg build nixosTests.lomiri

SuperSandro2000 commented on pkgs/by-name/pr/process-cpp/package.nix:
```diff
@@ -4,44 +4,64 @@
-    rev = "ee6d99a3278343f5fdcec7ed3dad38763e257310";
-    sha256 = "sha256-jDYXKCzrg/ZGFC2xpyfkn/f7J3t0cdOwHK2mLlYWNN0=";
+    rev = "7b0a829abcbcdd25d949e5f9e2c26bb985a58b31";
+    sha256 = "sha256-Az+lSJ7uVR4pAWvOeah5vFtIPb12eKp0nAFF1qsHZXA=";
```
We might as well use hash now

OPNA2608 replied:
Good point, applied.

SuperSandro2000 reviewed (commented):
Otherwise lgtm

OPNA2608 committed:
process-cpp: Refactor

- Switch to finalAttrs pattern
- Add pkgConfigModules & test
- Add updateScript
- Switch to src.hash

OPNA2608 committed:
process-cpp: unstable-2021-05-11 -> 3.0.1-unstable-2024-03-14

- Enable tests
- Enable strictDeps
- Add outputs
